src/opera/test/pge/dswx_s1/test_dswx_s1_pge.py

```diff
@@ -518,6 +568,127 @@ def test_dswx_s1_pge_input_validation(self):
 
         self._set_algorithm_parameters_path(runconfig_path, self.restore_test_algorithm_param_path)
 
+    def test_dswx_s1_pge_output_validation(self):
+        """Test the output validation checks made by DSWxS1PostProcessorMixin."""
+        runconfig_path = join(self.data_dir, 'test_dswx_s1_config.yaml')
+        test_runconfig_path = join(self.data_dir, 'invalid_dswx_s1_runconfig.yaml')
+
+        self._set_algorithm_parameters_path(runconfig_path, self.test_algorithm_parameters_path)
+
+        with open(runconfig_path, 'r', encoding='utf-8') as stream:
+            runconfig_dict = yaml.safe_load(stream)
+
+        primary_executable_group = runconfig_dict['RunConfig']['Groups']['PGE']['PrimaryExecutable']
+
+        # Set up an input directory empty of .tif files
+        burst_data = ()
+        self.generate_burst_output(burst_data, clear=True)
+
+        # Test with a SAS command that does not produce any output file,
+        # post-processor should detect that expected output is missing
+        primary_executable_group['ProgramPath'] = 'echo'
+        primary_executable_group['ProgramOptions'] = ['hello world']
+
+        with open(test_runconfig_path, 'w', encoding='utf-8') as config_fh:
+            yaml.safe_dump(runconfig_dict, config_fh, sort_keys=False)
+
+        try:
+            pge = DSWxS1Executor(pge_name="DSWxS1PgeTest", runconfig_path=test_runconfig_path)
+
+            with self.assertRaises(RuntimeError):
+                pge.run()
+
+            expected_output_file = 'dswx_s1_pge_test/output_dir/missing_dswx_s1.tif'
+            self.assertFalse(os.path.exists(expected_output_file))
```

Comment:
This test is not adding much, and could be removed

Reply:
I'll look at this again.  You're right the test is awkward.  I'll make it more clear what it's intended to do.  Thanks for the thorough review.